lib/net.js

```diff
@@ -319,14 +319,25 @@ Socket.prototype._onTimeout = function() {
 
 
 Socket.prototype.setNoDelay = function(enable) {
+  if (!this._handle) {
+    this.once('connect',
```

Comment:
It's actually documented as NOT being guaranteed: https://iojs.org/api/timers.html#timers_settimeout_callback_delay_arg